pilot/pkg/serviceregistry/kube/controller/network_test.go

```diff
@@ -79,12 +78,9 @@ func TestNetworkUpdateTriggers(t *testing.T) {
 	expectGateways := func(t *testing.T, expectedGws int) {
 		// wait for a notification
 		assert.ChannelHasItem(t, notifyCh)
-		retry.UntilSuccessOrFail(t, func() error {
```

Comment:
I tried this, and there were test failures when running it 100 times.

Reply:
Fixed, the reason is that the cb has not finished